spec/integ/matrix-to-irc.spec.js

```diff
@@ -624,6 +624,69 @@ describe("Matrix-to-IRC message bridging", function() {
         await p;
     });
 
+    it("should truncate multiline messages and include a full message URL", function(done) {
+        let tBody = "This\nis\na\nmessage\nwith\nmultiple\nline\nbreaks".split('\n');
+        let sdk = env.clientMock._client(config._botUserId);
+
+        sdk.uploadContent.and.returnValue(Promise.resolve("mxc://deadbeefcafe"));
+
+        env.ircMock._whenClient(roomMapping.server, testUser.nick, "say", (client, channel, text) => {
+            expect(client.nick).toEqual(testUser.nick);
+            expect(client.addr).toEqual(roomMapping.server);
+            expect(channel).toEqual(roomMapping.channel);
+            // don't be too brittle when checking this, but I expect to see the
+            // start of the first line and the mxc fragment
+            expect(text.indexOf(tBody[0])).toEqual(0);
+            expect(text.indexOf(tBody[1])).not.toEqual(0);
+            expect(text.indexOf('deadbeefcafe')).not.toEqual(-1);
+            done();
+        });
+
+        env.mockAppService._trigger("type:m.room.message", {
+            content: {
+                body: tBody.join("\n"),
+                msgtype: "m.text"
+            },
+            user_id: testUser.id,
```

Comment:
(legacy tests might use user_id, they should all be sender as the Matrix spec changed several years ago)

```suggestion
            sender: testUser.id,
```